File: src/src_platform_jfx/fmg/jfx/Cast.java
package fmg.jfx;

import java.util.List;

/**
 * Приведение типов от платформонезависемых читых Java классов fmg.common.geom.* к библиотечным javafx классам
 */
public final class Cast {

   public static javafx.geometry.Point2D     toPoint      (fmg.common.geom.Point       p) { return new javafx.geometry.Point2D    (     p.x     ,      p.y); }
   public static javafx.geometry.Point2D     toPoint      (fmg.common.geom.PointDouble p) { return new javafx.geometry.Point2D    (     p.x     ,      p.y); }
 //public static fmg.common.geom.Point       toPoint      (javafx.geometry.Point2D     p) { return new fmg.common.geom.Point      ((int)p.getX(), (int)p.getY()); }
   public static fmg.common.geom.PointDouble toPointDouble(javafx.geometry.Point2D     p) { return new fmg.common.geom.PointDouble(     p.getX(),      p.getY()); }

   public static javafx.geometry.Rectangle2D toRect      (fmg.common.geom.Rect        rc) { return new javafx.geometry.Rectangle2D(     rc.x        ,      rc.y        ,      rc.width     ,      rc.height); }
   public static javafx.geometry.Rectangle2D toRect      (fmg.common.geom.RectDouble  rc) { return new javafx.geometry.Rectangle2D(     rc.x        ,      rc.y        ,      rc.width     ,      rc.height); }
 //public static fmg.common.geom.Rect        toRect      (javafx.geometry.Rectangle2D rc) { return new fmg.common.geom.Rect       ((int)rc.getMinX(), (int)rc.getMinY(), (int)rc.getWidth(), (int)rc.getHeight()); }
   public static fmg.common.geom.RectDouble  toRectDouble(javafx.geometry.Rectangle2D rc) { return new fmg.common.geom.RectDouble (     rc.getMinX(),      rc.getMinY(),      rc.getWidth(),      rc.getHeight()); }

   public static javafx.geometry.Dimension2D  toSize      (fmg.common.geom.SizeDouble  size) { return new javafx.geometry.Dimension2D(     size.width     ,      size.height); }
   public static javafx.geometry.Dimension2D  toSize      (fmg.common.geom.Size        size) { return new javafx.geometry.Dimension2D(     size.width     ,      size.height); }
 //public static fmg.common.geom.Size         toSize      (javafx.geometry.Dimension2D size) { return new fmg.common.geom.Size       ((int)size.getWidth(), (int)size.getHeight()); }
   public static fmg.common.geom.SizeDouble   toSizeDouble(javafx.geometry.Dimension2D size) { return new fmg.common.geom.SizeDouble (     size.getWidth(),      size.getHeight()); }

   public static double[] toPolygon(fmg.common.geom.Region region, boolean xCoord) {
      return region.getPoints().stream().mapToDouble(p -> xCoord ? p.x : p.y).toArray();
   }

   public static double[] toPolygon(fmg.common.geom.RegionDouble region, boolean xCoord) {
      return region.getPoints().stream().mapToDouble(p -> xCoord ? p.x : p.y).toArray();
   }

   public static double[] toPolygon(List<fmg.common.geom.PointDouble> region, boolean xCoord) {
      return region.stream().mapToDouble(p -> xCoord ? p.x : p.y).toArray();
   }

   public static javafx.scene.paint.Color toColor(        fmg.common.Color clr) { return javafx.scene.paint.Color.rgb( clr.getR(), clr.getG(), clr.getB(), clr.getA() / 255.); }
   public static         fmg.common.Color toColor(javafx.scene.paint.Color clr) { return new fmg.common.Color(
         (int)(255*clr.getOpacity()),
         (int)(255*clr.getRed()),
         (int)(255*clr.getGreen()),
         (int)(255*clr.getBlue())); }

}

File: src/src_platform_ava/fmg/ava/utils/Cast.cs
using fmg.common;
using fmg.common.geom;

namespace fmg.uwp.utils {

   /// <summary>
   /// Приведение типов от платформо-независимых чистых C# классов fmg.common.* к библиотечным Avalonia классам
   /// </summary>
   public static class Cast {

      public static Size ToFmSize(this Avalonia.Size self) { return new Size((int)self.Width, (int)self.Height); }
      public static Avalonia.Size ToWinSize(this Size self) { return new Avalonia.Size(self.Width, self.Height); }

      public static SizeDouble ToFmSizeDouble(this Avalonia.Size self) { return new SizeDouble(self.Width, self.Height); }
      public static Avalonia.Size ToWinSize(this SizeDouble self) { return new Avalonia.Size(self.Width, self.Height); }

      public static Rect ToFmRect(this Avalonia.Rect self) { return new Rect((int)self.X, (int)self.Y, (int)self.Width, (int)self.Height); }
      public static Avalonia.Rect ToWinRect(this Rect self) { return new Avalonia.Rect(self.X, self.Y, self.Width, self.Height); }

      public static RectDouble ToFmRectDouble(this Avalonia.Rect self) { return new RectDouble(self.X, self.Y, self.Width, self.Height); }
      public static Avalonia.Rect ToWinRect(this RectDouble self) { return new Avalonia.Rect(self.X, self.Y, self.Width, self.Height); }

      public static Point ToFmPoint(this Avalonia.Point self) { return new Point((int)self.X, (int)self.Y); }
      public static Avalonia.Point ToWinPoint(this Point self) { return new Avalonia.Point(self.X, self.Y); }

      public static PointDouble ToFmPointDouble(this Avalonia.Point self) { return new PointDouble(self.X, self.Y); }
      public static Avalonia.Point ToWinPoint(this PointDouble self) { return new Avalonia.Point(self.X, self.Y); }

      public static Color ToFmColor(this Avalonia.Media.Color self) { return new Color(self.A, self.R, self.G, self.B); }
      public static Avalonia.Media.Color ToWinColor(this Color self) { return new Avalonia.Media.Color(self.A, self.B, self.G, self.R); }

      public static HSV ToHsvColor(this Avalonia.Media.Color self) { return new HSV(self.ToFmColor()); }
      public static Avalonia.Media.Color ToWinColor(this HSV self) { return self.ToColor().ToWinColor(); }

   }

}
